pkg/storage/export/export/vm-source.go

```diff
@@ -239,14 +254,15 @@ func (ctrl *VMExportController) updateVMExportVMStatus(vmExport *exportv1.Virtua
 
 	vmExportCopy := vmExport.DeepCopy()
 	vmExportCopy.Status.VirtualMachineName = pointer.P(vmExport.Spec.Source.Name)
+
 	if err := ctrl.updateCommonVMExportStatusFields(vmExport, vmExportCopy, exporterPod, service, sourceVolumes, getVolumeName); err != nil {
 		return requeue, err
 	}
+
 	if len(sourceVolumes.volumes) == 0 {
-		vmExportCopy.Status.Conditions = updateCondition(vmExportCopy.Status.Conditions, newReadyCondition(corev1.ConditionFalse, noVolumeVMReason, sourceVolumes.availableMessage))
 		vmExportCopy.Status.Phase = exportv1.Skipped
 	}
-	if !sourceVolumes.isPopulated {
+	if !sourceVolumes.isPopulated && sourceVolumes.readyCondition.Reason != vmNotFoundReason {
```

Comment:
I think we should re-queue if the VM is not found. The user asked for an export of a VM that doesn't exist yet, but could be there soon.

Reply:
Looks like we have a watch on VMs so we should get this for free